examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,407 @@
+                name=name,
+                system_message=system_message,
+            )
+        )
+
+    pro_agent = create_chat_agent("Pro",
+                                  pro_agent_config,
+                                  system_messages.messages[pro_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    con_agent = create_chat_agent("Con",
+                                  con_agent_config,
+                                  system_messages.messages[con_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    feedback_agent = create_chat_agent("Feedback",
+                                       feedback_agent_config,
+                                       FEEDBACK_AGENT_SYSTEM_MESSAGE)
+    logger.info("Pro, Con, and feedback agents created.")
+
+    # Determine user's side and assign user_agent and ai_agent based on user selection
+    agents = {"pro": (pro_agent, con_agent, "Pro", "Con"), "con": (con_agent, pro_agent, "Con", "Pro")}
+    user_agent, ai_agent, user_side, ai_side = agents[side]
+    logger.info(
+        f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+        f"LLM Delegate: {llm_delegate}"
+    )
+
+    logger.info(f"\n{user_side} Agent ({topic_name}):\n")
+
+    # Determine if the debate is autonomous or the user input for one side
+    if llm_delegate:
+        logger.info("Autonomous Debate Selected")
+        interactive_setting = False
+    else:
+        logger.info("Manual Debate Selected with an AI Agent")
+        interactive_setting = True
+        user_input: str = Prompt.ask(
+            "Your argument (or type 'f' for feedback, 'done' to end):"
+        )
+        user_agent.llm = None  # User message without LLM completion
+        user_agent.user_message = user_input
+
+    # Set up langroid tasks and run the debate
+    user_task = Task(user_agent, interactive=interactive_setting, restart=False)
+    ai_task = Task(ai_agent, interactive=False, single_round=True)
+    user_task.add_sub_task(ai_task)
+    if llm_delegate:
+        user_task.run(user_agent.user_message, turns=max_turns)
+    else:
+        user_task.run("get started", turns=max_turns)
+
+    # Determine the last agent based on turn count and alternation
+    # Note: user_agent and ai_agent are dynamically set based on the chosen user_side
+    last_agent = ai_agent if max_turns % 2 == 0 else user_agent
+
+    # Generate feedback summary and declare a winner using feedback agent
+    validation_message: List = last_agent.message_history
+    if not validation_message:
+        logger.warning("No agent message history found for the last agent")
+        validation_message = "No last agent history available to validate."
+
+    feedback_agent.enable_message(MetaphorSearchTool)
+    feedback_agent.enable_message(DoneTool)
+    feedback_agent_task = Task(feedback_agent, interactive=False)
+    feedback_agent_task.run(validation_message)
```

Comment:
I forgot to mention one thing here - the message history will contain the system message at the start, causing confusion to the feedback agent. You may want to add code to create a nicely annotated text from each side of the argument, by going thru the message_history. Something like

```
Pro: ... 

Con: ...

Pro:  ...

Con: ...
```

I'm adding a new experimental langroid feature later to langroid that adds a notion of an agent "listening" to one or more other agents and accumulating context so this type of manual stitching is avoided. This would be much like the bluesky social-network where people "follow" others. Your work made me realize there's this limitation in langroid that an agent can only get message from a **sinle** "parent" or peer task.
But that will be added later, so for now you'll need to this manually.

Reply:
Thanks! This was a good idea. Ready for review